controlplane/kubeadm/internal/cluster.go

```diff
@@ -185,7 +185,28 @@ func (m *ManagementCluster) RemoveEtcdMemberForMachine(ctx context.Context, clus
 		return err
 	}
 
-	return cluster.removeMemberForNode(ctx, machine.Status.NodeRef.Name)
+	controlPlaneNodes, err := cluster.getControlPlaneNodes(ctx)
+	if err != nil {
+		return err
+	}
+
+	nodeWillRemoved := machine.Status.NodeRef.Name
+	errs := []error{}
+	for _, node := range controlPlaneNodes.Items {
+		nodeForRemoving := node.Name
+		if nodeForRemoving == nodeWillRemoved {
```

Comment:
This breaks when we're removing the last member.

While I know the controller doesn't support removing the last member (i.e. scaling to zero), I don't think we should assume that constraint here.

The approach I had in mind was this: By default, choose the "to be deleted" node to create the client.  Then, iterate over the other nodes; choose the first one to create the client. That should support all cases. WDYT?